src/partition.jl

```diff
@@ -281,10 +281,19 @@ graph from the results of `partition`.
 A graph partitioning backend such as Metis or KaHyPar needs to be installed for this function to work
 if the subgraph vertices aren't specified explicitly.
 """
-function partition(
-  g::AbstractGraph; npartitions=nothing, nvertices_per_partition=nothing, kwargs...
-)
-  return partition(g, subgraph_vertices(g; npartitions, nvertices_per_partition, kwargs...))
+function partition(g::AbstractGraph; npartitions = nothing, nvertices_per_partition = nothing, sgraph_vertices = nothing, kwargs...)
+
+  if(count([npartitions != nothing, nvertices_per_partition != nothing, sgraph_vertices != nothing ]) >= 2)
```

Comment:
Could use:
```julia
if count(isnothing, (npartitions, nvertices_per_partition, , subgraph_vertices))
  [...]
end
```
Also I use the style that I don't wrap the check in parenthesis, i.e. please use `if expr` instead of `if(expr)`.

Reply:
Also, should the check be:
```julia
count(isnothing, (npartitions, nvertices_per_partition, subgraph_vertices)) != 1
```
i.e. you have to specify exactly one of them? What happens if you don't specify any of them?

Reply:
If you don't specify any then it will throw an error too from the call `partition(g, subgraph_vertices(g; npartitions, nvertices_per_partition, kwargs...))` which has its own error message built in. Good though I think to do the check 
`count(isnothing, (npartitions, nvertices_per_partition, subgraph_vertices)) != 1` in this function as it makes the most sense.